cmd/phpcs-server/main.go

```diff
@@ -0,0 +1,324 @@
+	"github.com/wptide/pkg/env"
+	"strconv"
+	"github.com/wptide/pkg/audit/ingest"
+	"github.com/wptide/pkg/audit/info"
+	"github.com/wptide/pkg/storage"
+	"github.com/wptide/pkg/storage/s3"
+	"flag"
+	"github.com/wptide/pkg/source"
+	"fmt"
+	"os"
+	"github.com/wptide/pkg/audit/phpcs"
+)
+
+var (
+	/** Compile time variables. **/
+	Version string // Set during build.
+	Build   string // Set during build.
+
+	/** Initialize interface instances **/
+	// Use the interface so that we can test.
+	TideClient tideApi.ClientInterface = &api.Client{}
+
+	// messageProvider is the primary source of messages to process.
+	messageProvider message.MessageProvider = sqs.NewSqsProvider(queueConfig.region, queueConfig.key, queueConfig.secret, queueConfig.queue)
+
+	// storageProvider is the cloud storage service for storing files.
+	storageProvider storage.StorageProvider = s3.NewS3Provider(s3Config.region, s3Config.key, s3Config.secret, s3Config.bucket)
+
+	// Temp folder for downloaded files.
+	tempFolder = env.GetEnv("PHPCS_TEMP_FOLDER", "/tmp")
+
+	/** Service Configurations **/
+	// Tide API configuration.
+	tideConfig = struct {
+		id           string
+		secret       string
+		authEndpoint string
+		host         string
+		protocol     string
+		version      string
+	}{
+		env.GetEnv("TIDE_API_KEY", ""),
+		env.GetEnv("TIDE_API_SECRET", ""),
+		env.GetEnv("TIDE_API_AUTH_URL", ""),
+		env.GetEnv("TIDE_API_HOST", ""),
+		env.GetEnv("TIDE_API_PROTOCOL", ""),
+		env.GetEnv("TIDE_API_VERSION", ""),
+	}
+
+	// SQS configuration.
+	queueConfig = struct {
+		region string
+		key    string
+		secret string
+		queue  string
+	}{
+		env.GetEnv("PHPCS_SQS_REGION", ""),
+		env.GetEnv("PHPCS_SQS_KEY", ""),
+		env.GetEnv("PHPCS_SQS_SECRET", ""),
+		env.GetEnv("PHPCS_SQS_QUEUE_NAME", ""),
+	}
+
+	// S3 configuration.
+	s3Config = struct {
+		region string
+		key    string
+		secret string
+		bucket string
+	}{
+		env.GetEnv("PHPCS_S3_REGION", ""),
+		env.GetEnv("PHPCS_S3_KEY", ""),
+		env.GetEnv("PHPCS_S3_SECRET", ""),
+		env.GetEnv("PHPCS_S3_BUCKET_NAME", ""),
+	}
+
+	/** Channels **/
+	// Number of concurrent audits.
+	bufferSize, _ = strconv.Atoi(env.GetEnv("PHPCS_CONCURRENT_AUDITS", "5"))
+	buffer        = make(chan struct{}, bufferSize)
+
+	// Create a channel that can terminate the app.
+	terminateChannel = make(chan struct{}, 1)
+
+	// Create a channel that receives messages from a queue.
+	cMessage chan *message.Message
+
+	/** Flags **/
+	bParseFlags = true
+
+	// Display versions?
+	bVersion = &[]bool{false}[0]
+
+	// A single URL to process. Will not poll queue.
+	flagUrl = &[]string{""}[0]
+
+	// Project visibility for URL. "private" or "public"
+	flagVisibility = &[]string{"private"}[0]
+
+	// The client login id in Tide API.
+	flagClient = &[]string{"wporg"}[0]
+
+	// Send to Stdout rather than API?
+	flagOutput = &[]string{""}[0]
+
+	/** Processes **/
+	// A slice of processes (in order) that need to be performed on a message.
+	processes = []audit.Processor{
```

Comment:
Does this mean we are no longer making the server config based? I only see two types of PHPCS audits supported or does this mean that we only support post processing for these two?